gpflow/conditionals.py

```diff
@@ -66,12 +76,13 @@
 
 
 def expand_independent_outputs(fvar, full_cov, full_cov_output):
+    # TODO point of this function?
     if not full_cov_output:
         # Output shape should be N x R or R x N x N, which it already is.
         return fvar
     elif full_cov:
         # Output shape should be ???
-        pass
+        raise NotImplementedError
```

Comment:
This function still needs to be implemented. In situations where we have independent outputs, we will always have a diagonal over P when full_cov_output is True. The intent for this function is that it reshapes from N x P to the appropriate shape.

Reply:
Yeah, but that's not something that's gonna happen in this branch...